js/src/utils/isSameShippingRate.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Checks if two shipping rates are the same.
+ *
+ * Shipping rates are same if:
+ *
+ * - they have the same defined id; or,
+ * - they have the same country and method.
+ *
+ * @param {Object} a Shipping rate A.
+ * @param {Object} b Shipping rate B.
+ * @return {boolean} True or false.
+ */
+const isSameShippingRate = ( a, b ) => {
+	return (
+		( a.id !== undefined && a.id === b.id ) ||
+		( a.country === b.country && a.method === b.method )
```

Comment:
❔ 

Could you explain me a bit more why is needed this part: `( a.country === b.country && a.method === b.method )` and why can we not rely only on the ids?

Reply:
So we are trying to compare two shipping rates, and one of them would be coming from the UI form state. Imagine a scenario like this: users load the page with existing shipping rates (the shipping rates would have the ids, since they are loaded from API). Those shipping rates would be put into form state, and the UI would depend on the form state. When users add a new shipping rate, it would be added into the form state, and it has **no ID**, since it has not been saved yet. So to be safe and defensive, we also check on the country and method combination.

Also, on the API side, we also checked on country and method combination to determine whether there are existing shipping rate: 

https://github.com/woocommerce/google-listings-and-ads/blob/4bfc1c723b7a80414b40bb578178a765ebaeed7b/src/API/Site/Controllers/MerchantCenter/ShippingRateController.php#L185

Reply:
Thanks, @ecgan for the explanation. 😃